src/ui/src/data/matchReview.ts

```diff
@@ -17,3 +17,41 @@ export async function getRecordMatch(id: string | null): Promise<RecordMatch> {
     );
   }
 }
+
+export async function linkRecordAndMatch(
+  id: string | null,
```

Comment:
Curious as to in what case the id will be null?  My thought is if the id is null, we'd expect an API error to be raised, so maybe we should raise a different error upstream from this call.

Reply:
Just because those types match the return types of the method that grabs the id from the url. I agree with `if the id is null, we'd expect an API error to be raised` and that is the expected behavior. The frontend client will still make the API call (HTTP request) with null as the ID. 

The client is expecting that the endpoint will return an error 400. That error response then will be passed to the page and the server error screen will display.

Let me know if what you are thinking is that the id as null is handled at the client level at which point we would not even start the HTTP request.

Reply:
example when id is null.
<img width="1319" alt="Screenshot 2025-05-05 at 2 49 38 PM" src="https://github.com/user-attachments/assets/d2eb0cc4-e317-4118-befa-79edb324be54" />

Reply:
A `POST /api/demo/record/null/link` will return a 422, not a 400.  Will it still be caught?

I see your reasoning here, since it's just passing on data from the original API that returned the id, if its null, its still an API error because the original GET call returned a value of null.  I guess its just an optimization in that case.

Reply:
yes, at this point all http errors (except the 404) will be handled in the page by presenting the server error screen to the user. The scenario of  calling `POST /api/demo/record/null/link` with a null id is not possible because the server error screen will not display the link and unlink buttons to the user. 

In another scenario. If the there is an error response from the POST /api/demo/record/[id]/link endpoint then the error is handled by the page and presented to the user as a toast notification. This error could be possible because we have issues during serialization or because the API server had some other issue while processing the request, but null id will not be the cause.

Reply:
@ericbuckley it could also be that a better experience would be to present the user with a different screen that instructs to go to the match queue page because a record was not properly selected, instead of displaying it as a server error.

We could create a ticket to follow up on this. LMK!

Reply:
I think its fine as a server error, this shouldn't ever happen and if it does, its likely because our API server is down for some reason.